src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/ReadsPipelineSpark.java

```diff
@@ -105,7 +110,10 @@ protected void runTool(final JavaSparkContext ctx) {
         }
 
         final JavaRDD<GATKRead> initialReads = getReads();
-        final JavaRDD<GATKRead> markedReadsWithOD = MarkDuplicatesSpark.mark(initialReads, getHeaderForReads(), duplicatesScoringStrategy, new OpticalDuplicateFinder());
+        if (parallelism == 0) {
+            parallelism = ReadsSparkSource.getRecommendedParallelism(ctx, getReadSourceName(), initialReads.partitions().size());
+        }
+        final JavaRDD<GATKRead> markedReadsWithOD = MarkDuplicatesSpark.mark(initialReads, getHeaderForReads(), duplicatesScoringStrategy, new OpticalDuplicateFinder(), parallelism);
```

Comment:
There may not be -- but if there isn't, the docs for the existing argument should describe it as the "shuffle and output parallelism" instead of just the "output parallelism"